src/executor.ts

```diff
@@ -80,45 +93,56 @@ export class Executor extends SmartContract {
    Player flips coin
    Smart contract verifies randomness from Oracle
   */
-
-  // TODO: add randomness element(s)
   @method
   flipCoin(
     player: PublicKey,
     stateBalance: Field,
     witness: MerkleMapWitness,
     channelDeltaBalance: Int64,
     channelNonce: Field,
-    channelBalanceSignature: Signature
-  ): Int64 {
+    channelBalanceSignature: Signature,
+    randomnessSignature: Signature,
+    encryptionCT1: Field,
+    encryptionCT2: Field,
+    encryptionGroup: Group,
+    executorPrivateKey: PrivateKey
+  ): [Int64, Field[]] {
     this.proveState(player, stateBalance, witness);
+    const oraclePublicKey = this.oraclePublicKey.get();
+    this.oraclePublicKey.assertEquals(oraclePublicKey);
+
     let channelBalanceProof = new ChannelBalanceProof(player, this.address);
     channelBalanceProof.deltaBalance = channelDeltaBalance;
     channelBalanceProof.nonce = channelNonce;
     channelBalanceProof.player.assertEquals(player);
     channelBalanceProof.verify(channelBalanceSignature).assertTrue();
 
-    // console.log(`In the method - before: ${channelBalanceProof.toString()}`);
-
     let trueBalance = channelBalanceProof.deltaBalance.add(
       Int64.fromField(stateBalance)
     );
 
-    // TODO: just using 100 as a random value for now, clean this up
-    const isValidFlip = Circuit.if(
-      trueBalance.isPositive(),
-      (() => trueBalance.toField().gt(100))(),
-      (() => Bool(false))()
+    const isValidFlip = trueBalance.toField().gt(Field(25));
```

Comment:
Cut the user off if their IOU is only good for 5 more flips.  This is totally arbitrary but felt like a reasonable way to avoid being frontrun somehow with multiple flips idk...